spec/gateways/s3_spec.rb

```diff
@@ -8,4 +8,26 @@
   it 'uploads the data' do
     expect(subject.upload(data: data)).to eq({})
   end
+
+  context 'given we read from the S3 file' do
+    before do
+      Rails.application.config.s3_aws_config = {
+        stub_responses: {
+          get_object: ->(context) {
+            if context.params.fetch(:bucket) == bucket && context.params.fetch(:key) == key
+              { body: 'some data' }
+            end
+          }
+        }
+      }
+    end
+
+    after do
+      Rails.application.config.s3_aws_config = { stub_responses: true }
```

Comment:
it might be better to restore this to the actual original value, to reduce duplication and potential hidden breakages.

```ruby
around do |example|
  original_s3_aws_config = Rails.application.config.s3_aws_config
  begin
    example.run
  ensure
    Rails.application.config.s3_aws_config = original_s3_aws_config
    # set config
  end
end
```